File: HW8/Palindrome.py
def longest_palindrome(s):
    frequency = [0] * 128
    length = 0
    for c in s:
        frequency[ord(c)] += 1
        if frequency[ord(c)] == 2:
            length += 2
            frequency[ord(c)] = 0
    if length < len(s):
        length += 1
    return length

if __name__ == "__main__":
    s = "aabbccdd"

    # Function call
    print longest_palindrome(s)

File: HW8/Palindrome.java
public class Palindrome {

	public static int longestPalindrome(String s) {
		int[] frequency = new int[128];
		int len = 0;
		for (char c : s.toCharArray()) {
			frequency[c]++;
			if (frequency[c] == 2) {
				len += 2;
				frequency[c] = 0;
			}
		}
		if (len < s.length()) {
			len++;
		}
		return len;
	}

	public static void main(String[] args) {
		String s = "aAb";

		// Function call
		System.out.print(longestPalindrome(s));
	}
}
